agent/agent.go

```diff
@@ -754,6 +755,11 @@ func (a *Agent) Start(ctx context.Context) error {
 			return fmt.Errorf("can't start agent: client agents are not supported with v2 resources")
 		}
 
+		// the conn is used to connect to the consul server agent
+		conn, err := a.baseDeps.GRPCConnPool.ClientConn(a.baseDeps.RuntimeConfig.Datacenter)
```

Comment:
I believe this outbound connection is going to be sent over the yamux port, and not the grpc / grpc_tls ports. I'm unfamiliar with whether this will pose any security issues (due to servers blindly trusting the incoming request over that port, etc). I think we should get feedback from @mkeeler on whether this is a non-issue before merging.